include/inexor/vulkan-renderer/wrapper/mesh_buffer.hpp

```diff
@@ -26,60 +28,95 @@ namespace inexor::vulkan_renderer::wrapper {
                const std::size_t number_of_indices, void *indices);
 
-    /// @brief Creates a vertex buffer without index buffer.
-    MeshBuffer(const wrapper::Device &device, VkQueue data_transfer_queue,
+    /// @brief Creates a new vertex buffer with an associated index buffer but does not copy memory into it.
+    /// This is useful when you know the size of the buffer but you don't know it's data values yet.
+    MeshBuffer(const wrapper::Device &device, const VkQueue data_transfer_queue,
+               const std::uint32_t data_transfer_queue_family_index, const VmaAllocator vma_allocator,
+               const std::string &name, const VkDeviceSize size_of_vertex_structure,
+               const std::size_t number_of_vertices, const VkDeviceSize size_of_index_structure,
+               const std::size_t number_of_indices);
+
+    /// @brief Creates a vertex buffer without index buffer, but copies the vertex data into it.
+    MeshBuffer(const wrapper::Device &device, const VkQueue data_transfer_queue,
                const std::uint32_t data_transfer_queue_family_index, const VmaAllocator vma_allocator,
                const std::string &name, const VkDeviceSize size_of_vertex_structure,
                const std::size_t number_of_vertices, void *vertices);
-    MeshBuffer(const MeshBuffer &) = delete;
-    MeshBuffer(MeshBuffer &&) noexcept;
-    ~MeshBuffer();
 
-    MeshBuffer &operator=(const MeshBuffer &) = delete;
-    MeshBuffer &operator=(MeshBuffer &&) = default;
+    /// @brief Creates a vertex buffer without index buffer and copies no vertex data into it.
+    MeshBuffer(const wrapper::Device &device, const VkQueue data_transfer_queue,
+               const std::uint32_t data_transfer_queue_family_index, const VmaAllocator vma_allocator,
+               const std::string &name, const VkDeviceSize size_of_vertex_structure,
+               const std::size_t number_of_vertices);
+
+    ~MeshBuffer();
 
-    [[nodiscard]] VkBuffer vertex_buffer() const {
+    [[nodiscard]] VkBuffer get_vertex_buffer() const {
         return m_vertex_buffer.buffer();
     }
 
     [[nodiscard]] bool has_index_buffer() const {
-        return m_index_buffer.has_value();
+        return (m_index_buffer) ? true : false;
     }
 
-    [[nodiscard]] std::optional<VkBuffer> index_buffer() const {
-        assert(m_index_buffer);
-        return m_index_buffer->buffer();
+    [[nodiscard]] VkBuffer get_index_buffer() const {
+        if (!m_index_buffer) {
+            throw std::runtime_error(std::string("Error: No index buffer for mesh " + m_name + "!"));
+        }
+
+        return m_index_buffer.value().buffer();
     }
 
-    [[nodiscard]] const std::uint32_t vertex_count() const {
+    [[nodiscard]] const std::uint32_t get_vertex_count() const {
         return m_number_of_vertices;
     }
 
-    [[nodiscard]] const std::uint32_t index_cound() const {
+    [[nodiscard]] const std::uint32_t get_index_count() const {
```

Comment:
Returning const is useless here.

Reply:
I removed const for `get_index_count` and `get_vertex_count`.
Done.